ethereum/forge-test/GasBenchmark.t.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import "forge-std/Test.sol";
+
+import "@pythnetwork/pyth-sdk-solidity/IPyth.sol";
+import "@pythnetwork/pyth-sdk-solidity/PythStructs.sol";
+import "./utils/WormholeTestUtils.t.sol";
+import "./utils/PythTestUtils.t.sol";
+
+contract GasBenchmark is Test, WormholeTestUtils, PythTestUtils {
+    // 19, current mainnet number of guardians, is used to have gas estimates
+    // close to our mainnet transactions.
+    uint8 constant NO_GUARDIANS = 19;
+    uint8 constant NO_SIGNERS = 13;
+
+    // We use 5 prices to form a batch of 5 prices, close to our mainnet transactions.
+    uint8 constant NO_PRICES = 5;
+
+    uint constant BENCHMARK_ITERATIONS = 50;
+
+    IPyth public pyth;
+    
+    bytes32[] priceIds;
+    PythStructs.Price[] prices;
+    uint64 sequence;
+
+    function setUp() public {
+        pyth = IPyth(setUpPyth(setUpWormhole(NO_GUARDIANS)));
+
+        priceIds = new bytes32[](NO_PRICES);
+        priceIds[0] = bytes32(0x1000000000000000000000000000000000000000000000000000000000000f00);
+        for (uint i = 1; i < NO_PRICES; ++i) {
+            priceIds[i] = bytes32(uint256(priceIds[i-1])+1);
+        }
+
+        for (uint i = 0; i < NO_PRICES; ++i) {
+            prices.push(PythStructs.Price(
+                100, // Price
+                10, // Confidence
+                -5, // Expo
+                1 // publishTime
```

Comment:
It might be worth randomising these (and `advancePrices`), just to make these benchmarks more realistic.

Reply:
Done.